### PR TITLE
Add AWS Native 3rd party extension guide

### PR DESCRIPTION
**Pull request by danielrbradley**
## Description

Add a new section to the AWS Native Cloud Control guide on how to use the ExtensionResource to access third party resources.

This was part of #4096 but that's now not being merged for the time being.

### CODE REVIEW

File: themes/default/content/registry/packages/aws-native/_index.md
Comment (t0yv0) [inline themes/default/content/registry/packages/aws-native/_index.md:273]: Can we randomize the name? If I try this it says bucket already exists, which is likely.

Comment (t0yv0) [reply themes/default/content/registry/packages/aws-native/_index.md:273]: After updating the resource

```
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: f5ee580c-289f-47ec-ad3b-57ee50131f75, api error ValidationException: 2 validation errors detected: Value '' at 'typeName' failed to satisfy constraint: Member must have length greater than or equal to 10; Value '' at 'typeName' failed to satisfy constraint: Member must satisfy regular expression pattern: [A-Za-z0-9]{2,64}::[A-Za-z0-9]{2,64}::[A-Za-z0-9]{2,64}

```

**Review by t0yv0 — CHANGES_REQUESTED** [review]
The examples don't work for me.

File: themes/default/content/registry/packages/aws-native/_index.md
Comment (danielrbradley) [reply themes/default/content/registry/packages/aws-native/_index.md:273]: Randomising the name is the exact purpose of this PR which has been waiting on an approval for 3 weeks:
- https://github.com/pulumi/pulumi-aws-native/pull/1453

I'm not too sure where those validation errors are coming from ... this passed for me when testing.

Comment (t0yv0) [reply themes/default/content/registry/packages/aws-native/_index.md:273]: Ah I see, well the bucket naming issue is probably fine the users will figure it out, but should this like be typeName instead type? Why do I hit the error? I can try again.

Comment (t0yv0) [reply themes/default/content/registry/packages/aws-native/_index.md:273]:
```python
import pulumi
import pulumi_aws_native as aws_native

my_bucket = aws_native.ExtensionResource("myBucket",
    type="AWS::S3::Bucket",
    properties={
        "BucketName": "my-bucket-t0yv0-2024apr18",
    })

```

This worked on `pulumi_aws_native  0.102.0`, perhaps I was fat-fingering something yesterday.

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**
